pkg/core/resources.go

```diff
@@ -18,15 +18,19 @@ type (
 	// Construct describes a resource at the source code, Klotho annotation level
 	Construct interface {
 		BaseConstruct
-		// Provenance returns the AnnotationKey that the construct was created by
-		Provenance() AnnotationKey
+
+		// AnnotationCapability returns the annotation capability of the construct. This helps us tie the annotation types to the constructs for the time being
+		AnnotationCapability() string
 	}
 
+	// BaseConstructsRefSet is a set of BaseConstructs
```

Comment:
```suggestion
```
nit: pretty clear what this is - a godoc which reiterates its name isn't necessary.

Reply:
will remove